lib/web/websocket/websocket.js

```diff
@@ -666,7 +668,7 @@ class WebSocket extends EventTarget {
     let code = 1005
     let reason = ''
 
-    const result = this.#parser.closingInfo
+    const result = this.#parser?.closingInfo
```

Comment:
why this change?

Reply:
`WebSocket.#socketClose` will thrown an error trying to access properties on `undefined` as `WebSocket.#parser` is `undefined` when the connection never establishes.

Reply:
seems odd

Reply:
https://github.com/nodejs/undici/blob/89a46dd54f7d7db7513c435fac8042769ee9e9b5/lib/web/websocket/websocket.js#L481-L490

`WebSocket.#parser` is set inside `WebSocket.#onConnectionEstablished`, which is never reached since an actual connection is never established.

Reply:
i checked it and it seems ok.